include/can_interface.h

```diff
@@ -476,6 +641,326 @@ class CANTXMessage : public ICANTXMessage
+#endif
+    }
+
+    template <typename... Ts>
+    /**
+     * @brief Construct a new MultiplexedCANTXMessage object, for use if your message uses a multiplexor, and
+     * automatically adds it to a VirtualTimerGroup, default to standard id
+     *
+     * @param can_interface The ICAN object the message will be transmitted on
+     * @param id The ID of the CAN message
+     * @param length The length in bytes of the message
+     * @param period The transmit period in ms of the message
+     * @param timer_group A timer group to add the transmit timer to
+     * @param multiplexor_values_to_transmit An array of multiplexor values that gets cycled through to automatically
+     * transmit the messages
+     * @param multiplexor The CANSignal that will be used to multiplex the message
+     * @param signal_groups The SignalGroups contained in the message
+     */
+    MultiplexedCANTXMessage(ICAN &can_interface,
+                            uint32_t id,
+                            uint8_t length,
+                            uint32_t period,
+                            VirtualTimerGroup &timer_group,
+                            std::array<MultiplexorType, num_multiplexors_to_transmit> multiplexor_values_to_transmit,
+                            ITypedCANSignal<MultiplexorType> &multiplexor,
+                            Ts &... signal_groups)
+        : MultiplexedCANTXMessage(can_interface,
+                                  id,
+                                  false,
+                                  length,
+                                  period,
+                                  timer_group,
+                                  multiplexor_values_to_transmit,
+                                  multiplexor,
+                                  signal_groups...)
+    {
+    }
+
+    void EncodeAndSend() override  // increments multiplexor automatically
+    {
+        *multiplexor_ = multiplexor_values_to_transmit_.at(static_cast<size_t>(multiplexor_index_));
+        EncodeSignals(GetSignalGroupIndex(*multiplexor_));
+        can_interface_.SendMessage(message_);
+        if (multiplexor_index_ < num_multiplexors_to_transmit - 1)
+        {
+            multiplexor_index_ += 1;
+        }
+        else
+        {
+            multiplexor_index_ = 0;
+        }
+    }
+
+    void EncodeAndSend(MultiplexorType multiplexor_value)
+    {
+        *multiplexor_ = multiplexor_value;
+        EncodeSignals(GetSignalGroupIndex(*multiplexor_));
+        can_interface_.SendMessage(message_);
+    }
+
+    uint32_t GetID() override { return message_.id_; }
+
+#if !defined(NATIVE)  // workaround for unit tests
+    VirtualTimer &GetTransmitTimer() override { return transmit_timer_; }
+#endif
+
+    void Enable() { transmit_timer_.Enable(); }
+    void Disable() { transmit_timer_.Disable(); }
+
+private:
+    ICAN &can_interface_;
+    CANMessage message_;
+#if !defined(NATIVE)  // workaround for unit tests
+    VirtualTimer transmit_timer_;
```

Comment:
Since L820 and L821 uses transmit timer, I think unit tests need this line to pass. Can remove the if def